v5 - Add formatting checks to Python linting process

alexandreNoens opened the pull request:
Ensure `make lint-python` also checks code formatting, not just lint rules. This helps maintain consistent style across files and avoids formatting discrepancies.

Resolves: #23

alexandreNoens committed:
Add formatting checks to Python linting process

Ensure `make lint-python` also checks code formatting, not just lint
rules. This helps maintain consistent style across files and avoids
formatting discrepancies.

Resolves: #23

gentooboontoo commented:
Mob : à tester @gentooboontoo @SajedehAdelia

gentooboontoo reviewed (approved):
It works fine :+1:

gentooboontoo commented:
Suggestions to improve this Python linting target in another commit (backport from v3-legacy branch if included):

- check if `TEMPLATE_DIRS` exists 
- `--no-run-if-empty` added as argument to `xargs` to prevent failure when there is no template files (empty list)

```diff
diff --git a/base.Makefile b/base.Makefile
index fe77d62..c7dbeff 100644
--- a/base.Makefile
+++ b/base.Makefile
@@ -326,7 +326,10 @@ lint-pytho%: ## lint-python: Lint python source
        $(LOG)
        $(RUFF) check $(LINTED_PYTHON_DIRS)
        $(RUFF) format --diff $(LINTED_PYTHON_DIRS)
-       find $(TEMPLATE_DIRS) -type f -name '*.jinja2' | xargs $(DJLINT) --check --lint
+       if [ -d $(TEMPLATE_DIRS) ]; then \
+               find $(TEMPLATE_DIRS) -type f -name '*.jinja2' | \
+               xargs --no-run-if-empty $(DJLINT) --check --lint; \
+       fi

 lint-nod%: ## lint-node: Lint node source
        $(LOG)
```

alexandreNoens committed:
Add a condition for the fix and lint Python commands

Some projects don't have a Jinja2 template, so there's no need to use
djLint to format them.

SajedehAdelia reviewed (approved):
I checked the fix-23 branch and compered it with branch v5 in teepy. It functions as it should. I approve ^^

alexandreNoens committed:
Improve the printout

Now it displays the executed command.
And I used a variable to store the djlint command.

alexandreNoens committed:
Fix assignment operator

This ensures the command is expanded once at definition time,
which avoids unnecessary repeated evaluations.

glepretre reviewed (commented):
Good move :+1:
